src/e2e/utils/helpers.ts

```diff
@@ -160,3 +162,63 @@ export function removeUnicodeChars(text: string): string {
   // eslint-disable-next-line no-control-regex
   return text.replace(/[^\x00-\x7F]/g, "");
 }
+
+export const clickOnATagCheckDomain = async (
+  aTag: Locator,
+  host: string | RegExp = /.*/,
+  path: string | RegExp = /.*/,
+  page: Page,
+) => {
+  if (typeof host === "string")
+    host = new RegExp(escapeRegExp(host.replace(/^(https?:\/\/)/, "")));
+  if (typeof path === "string") path = new RegExp(".*" + path + ".*");
+
+  const href = await aTag.getAttribute("href");
+  if (href === null) return false;
+
+  await page.goto(href);
+  const currentUrl = new URL(page.url());
+  const perceivedHost = currentUrl.hostname;
+  const perceivedPath = currentUrl.pathname;
+  expect(host.test(perceivedHost)).toBeTruthy();
+  expect(path.test(perceivedPath)).toBeTruthy();
+  await page.goBack();
+};
+
+export const escapeRegExp = (str: string): string => {
+  return str.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+};
+
+export const forceLoginAs = async (
+  email: string,
+  password: string,
+  page: Page,
+  landingPage: LandingPage,
+  authPage: AuthPage,
+) => {
+  test.slow(
+    true,
+    "this test runs through the welcome scan flow, increasing timeout to address it",
+  );
+
+  // speed up test by ignoring non necessary requests
+  await page.route(/(analytics)/, async (route) => {
+    await route.abort();
+  });
+  await page.context().clearCookies();
+  await landingPage.open();
+  await landingPage.goToSignIn();
+  let visible = true;
+  try {
+    await expect(authPage.useDifferentEmailButton).toBeVisible();
+  } catch {
+    visible = false;
+  }
+  if (visible) {
+    await authPage.useDifferentEmailButton.click();
+    await page.waitForURL(/^(?!.*signin).*/);
+  }
+  await authPage.signIn(email, password);
+  await page.waitForURL("**/user/dashboard");
+  expect(page.url()).toContain("/user/dashboard");
```

Comment:
Any benefits to using `expect(page.url()).toContain(...)` vs [`await expect(page).toHaveURL(/.../);`](https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-url) for URL assertions?

Reply:
Sure ill change it.